geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerFeatureSource.scala

```diff
@@ -65,7 +68,14 @@ class KafkaConsumerFeatureSource(entry: ContentEntry,
     }
   }
 
-  val features = Maps.newConcurrentMap[String, FeatureHolder]()
+  val features = if (expiry) {
+    CacheBuilder
+      .newBuilder()
+      .expireAfterWrite(expirationPeriod, TimeUnit.MILLISECONDS)
+      .build()
+  } else {
+    Maps.newConcurrentMap[String, FeatureHolder]()
```

Comment:
To simplify, I would just make it always use a cache. If there is no expiry, then don't use the expireAfterWrite option. But then you don't need all the duplicate code below for the 2 paths, plus all the 'asInstanceOf' (which you should usually try to avoid).